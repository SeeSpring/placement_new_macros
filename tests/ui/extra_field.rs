use placement_new_macros::init;
use std::mem::MaybeUninit;

fn main() {
    struct T {
        t: u32,
    }
    let mut v = MaybeUninit::<T>::uninit();
    // missing field `t`
    init!(v, T { t: 0, s: 0 });
}
