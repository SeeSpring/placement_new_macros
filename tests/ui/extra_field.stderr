error[E0560]: struct `main::T` has no field named `s`
  --> $DIR/extra_field.rs:10:24
   |
10 |     init!(v, T { t: 0, s: 0 });
   |                        ^ `main::T` does not have this field
   |
   = note: available fields are: `t`

error[E0609]: no field `s` on type `main::T`
  --> $DIR/extra_field.rs:10:24
   |
10 |     init!(v, T { t: 0, s: 0 });
   |                        ^ help: a field with a similar name exists: `t`
